### PR TITLE
Feat/1285 accordion and accordion group

### LINKED ISSUE #1285: Be able to use the Accordion component from Felles Designsystem
### Description

We want to be able to display information with the use of the `Accordion` component from `@digdir/design-system-react`. 

## Proposed techincal solution
~~This needs to be solved as a new component~~ ~~Can be solved by providing an option to `Group` and `RepeatingGroup` which implements the `Accordion` component from `@digdir/design-system-react`.~~

~~The new component behaviour of `Group` and `RepeatingGroup` should:~~
- ~~Be able to accept a list of options to support multiple items in the `Accordion`.~~ 
- ~~Each option should accept a `title` text resource via the `textResourceBindings`-property.~~ 
- ~~Be able to accept dynamic content. Text body should be provided by placing a  `Paragraph` inside the `Accordion`.~~

After discussing this with the apps team, we have decided that we will introduce two new components. `Accordion` and `AccordionGroup`. 

## Accordion
`Accordion` would be a single stand alone component which can have a title and some contents in a body. The contents of the body should be comprised of other Altinn 3 components. If the `Accordion` should contain only text, this should be solved with placing a `Paragraph` inside it. There will be some contraints on what content could go into an `Accordion`. The title of the `Accordion` will be provided via `textResourceBindings`. 

Closed state: 
image

Open state: 
image

## AccordionGroup
`AccordionGroup` should be a component that groups `Accordion` components together. The only content that would be allowed to place inside an `AccordionGroup` is `Accordions`. 

image

### Additional Information

[The accordion component from digdir's design system](https://digdir.github.io/designsystem/?path=/docs/kjernekomponenter-accordion--docs)

### PR DESCRIPTION
**Pull request by mikaelrss**
## Description

Adds two new components, `Accordion` and `AccordionGroup`.

## Related Issue(s)

- closes #1285 

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [x] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [x] Has been added/updated - [Added a PR here](https://github.com/Altinn/altinn-studio-docs/pull/1024)
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Changes/additions to component properties
  - [x] Changes are reflected in both `src/layout/layout.d.ts` and `layout.schema.v1.json`, and these are all backwards-compatible
  - [ ] No changes made
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by bjosttveit** [comment]
Does this solve https://github.com/Altinn/app-frontend-react/issues/165?

**Comment by mikaelrss** [comment]
I guess it could solve #165. As of right now, the `Accordion` only supports `Paragraph` and `Button` components as children, and it seems like the issue needs form components to be placed inside an `Accordion`. I guess we could expand to accompany for more components.

File: schemas/json/layout/layout.schema.v1.json [hunk 1]
```diff
@@ -160,6 +162,34 @@
...
+        }
+      },
+      "required": ["children"]
+    },
+    "accordionGroupComponent": {
+      "properties": {
+        "children": {
+          "title": "Children",
+          "type": "array",
+          "description": "An array of the \"id\" of child components belonging to the button group.",
```
Comment (olemartinorg) [inline schemas/json/layout/layout.schema.v1.json:184]: You should probably update these descriptions 😉

File: src/layout/AccordionGroup/hierarchy.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  /**
+   * Check if a component can be rendered in an AccordionGroup.
+   */
+  private canRenderInAccordionGroup(generator: HierarchyGenerator, childId: string, outputWarning = true): boolean {
+    const prototype = generator.prototype(childId);
+    const def = prototype && generator.getLayoutComponentObject(prototype.type);
+
+    if (outputWarning && prototype && !def?.canRenderInAccordionGroup()) {
+      console.warn(
```
Comment (olemartinorg) [inline src/layout/AccordionGroup/hierarchy.ts:61]: You should probably use `window.logWarn` instead, to make this appear in our DevTools log. `console.log` etc is for temporary debugging messages to _us_, and should never be committed, but messages to application developers should be sent to their DevTools.

**Review by olemartinorg — APPROVED** [review]

**Comment by mikaelrss** [comment]
> But right now there's probably no coverage over your hiearchy generator etc, so maybe a full integration test for these components are in order? It might be difficult to do with pure unit tests, but easy to observe the effects of it in a Cypress test.

I have already added one cypress test testing that the accordion opens and closes, and that the content is invisible/visible. This should cover `Accordion/hierarchy.ts` right? It also covers `AccordoinGroup/hierarchy.ts` since `frontend-test` includes an accordion group as well. However, I haven't made any assertions to specifically test `AccordionGroup`, as this is mainly a style change for `Accordion`s.  

@olemartinorg Do you think this is enough coverage, or is there something I am missing?

**Comment by olemartinorg** [comment]
@mikaelrss Sorry for missing the test! 🙈 Yup, sounds like more than enough tests, then. 🙏 

@bjosttveit Ah! I changed something in `frontend-test` a while ago that might affect that. In the text resource, we didn't escape the `2. ` so markdown formatting thought it a numbered list item. That might explain parts of it, I think. It seems to me that Percy lags behind a while after there has been a change, so I guess we should just expect there to be large diffs a while after a large change... :-/

**Pull request merged**